controllers/sandbox_controller.go

```diff
@@ -114,13 +127,48 @@ func (r *SandboxReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	// Update status
 	if statusUpdateErr := r.updateStatus(ctx, oldStatus, sandbox); statusUpdateErr != nil {
 		// Surface update error
+		statusUpdateErr = fmt.Errorf("faild to update status: %w", statusUpdateErr)
 		err = errors.Join(err, statusUpdateErr)
 	}
 
+	if recordErr := r.recordFirstReadyMetric(ctx, sandbox); recordErr != nil {
```

Comment:
Could there be a benign race condition between the `patch` that happens here inside the `r.recordFirstReadyMetric` and the `r.updateStatus` from above which will update the resourceVersion ? I.e. two separate writes to the same object in the same reconcile loop. This could make `sandbox_creation_latency` being recorded twice one when the error happens and then when the requeue because of the error, happens. We could fix this by serializing the writes.